### PR TITLE
Replace vfsgen with embed.FS

### LINKED ISSUE #29677: profile diff cannot compare two profiles directly unless `--manifests manifests` is set
(**NOTE**: This is used to report product bugs:
  To report a security vulnerability, please visit <https://istio.io/about/security-vulnerabilities>
  To ask questions about how to use Istio, please visit <https://discuss.istio.io>)

**Bug description**
The correct/workable usages of profile diff from a downloaded istio release are:
(1)
```
istioctl profile diff preview default
```
(2)
```
istioctl profile diff --manifests manifests default.yaml demo.yaml
```

but when I following the istio.io instruction: https://istio.io/latest/docs/reference/commands/istioctl/#istioctl-profile-diff which says, this command can compare two profiles yaml directly by passing them to the args, which is not working:

when i run:
```
istioctl profile diff manifests/profiles/default.yaml manifests/profiles/demo.yaml
```
it returns:
```
Error: could not read "manifests/profiles/default.yaml": Asset profiles/manifests/profiles/default.yaml.yaml not found
```

We should fix possible bugs here and add an explanation in the flags help text for this command to make it clear to the users also developers.

cc @ostromart @esnible @howardjohn 

[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Extensions and Telemetry
[ ] Security
[ ] Test and Release
[x ] User Experience
[ ] Developer Infrastructure
[ ] Upgrade

**Expected behavior**

**Steps to reproduce the bug**

**Version** (include the output of `istioctl version --remote` and `kubectl version --short` and `helm version --short` if you used Helm)

**How was Istio installed?**

**Environment where the bug was observed (cloud vendor, OS, etc)**

Additionally, please consider running `istioctl bug-report` and attach the generated cluster-state tarball to this issue.
Refer [cluster state archive](http://istio.io/help/bugs/#generating-a-cluster-state-archive) for more details.

### PR DESCRIPTION
**Pull request by howardjohn**
Blocked by making golang 1.16 a requirement to build Istio

Fixes https://github.com/istio/istio/issues/29677

### CODE REVIEW

**Comment by howardjohn** [comment]
Lint failure is a bug, IMO: https://github.com/golang/lint/issues/503

File: operator/pkg/helm/renderer.go [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+
+// RenderManifest renders the current helm templates with the current values and returns the resulting YAML manifest string.
+func (h *Renderer) RenderManifest(values string) (string, error) {
+	if !h.started {
+		return "", fmt.Errorf("fileTemplateRenderer for %s not started in renderChart", h.componentName)
+	}
+	return renderChart(h.namespace, values, h.chart)
+}
+
+func GetFilesRecursive(f fs.FS, root string) ([]string, error) {
```
Comment (ostromart) [inline operator/pkg/helm/renderer.go:77]: we have util.FindFiles, would that work here?

**Comment by ostromart** [comment]
overall looks great, thanks for doing this.

File: Makefile.core.mk [hunk 2]
```diff
@@ -342,22 +342,20 @@ go-gen:
...
 	@REFRESH_GOLDEN=true go test ${GOBUILDFLAGS} ./operator/...
 	@REFRESH_GOLDEN=true go test ${GOBUILDFLAGS} ./pkg/kube/inject/...
 	@REFRESH_GOLDEN=true go test ${GOBUILDFLAGS} ./pilot/pkg/security/authz/builder/...
 
 update-golden: refresh-goldens
 
-gen: mod-download-go go-gen mirror-licenses format update-crds operator-proto sync-configs-from-istiod gen-kustomize update-golden ## Update all generated code.
+# Keep dummy target since some build pipelines depend on this
+gen-charts:
+	@echo "This target is no longer required and will be removed in the future"
```
Comment (dgn) [inline Makefile.core.mk:354]: nit: maybe "FIXME: the gen-charts target is no longer..."

File: operator/scripts/create_release_charts.sh [hunk 3]
```diff
@@ -1,49 +0,0 @@
-#!/usr/bin/env bash
```
Comment (dgn) [inline operator/scripts/create_release_charts.sh:1]: where's this called? I don't see a reference to it being removed anywhere

**Review by dgn — COMMENTED** [review]
Thanks, this is great. What other changes are needed to get this rolling? I assume updating build containers with golang 1.16? As the tests are passing I'm guessing that's happened already

**Review by morvencao — COMMENTED** [review]
Do we still need these lines? The `global.yaml` in the `charts` directory was removed: https://github.com/istio/istio/blob/670075ebe7cc2e7ec15d39edca343d05c26aa6b4/operator/scripts/update_version.sh#L24-L26

**Comment by howardjohn** [comment]
> Thanks, this is great. What other changes are needed to get this rolling? I assume updating build containers with golang 1.16? As the tests are passing I'm guessing that's happened already

This is pretty much it! We are already on 1.16. This will be the first 1.16 feature, so we will need to broadcast to devs to upgrade if they haven't. They can also use the build container. Third party distributors also need to update their build pipelines if they have custom ones, but this won't actually make it into a real release (1.10) for 2+ months so lots of tyime.

**Review by ostromart — APPROVED** [review]

**Review by dgn — APPROVED** [review]

File: operator/scripts/create_release_charts.sh (hunk 3 above)
Comment (howardjohn) [reply operator/scripts/create_release_charts.sh:1]: From operator/scripts/create_assets_gen.sh from gen-charts

**Review by nmittler — APPROVED** [review]

**Pull request merged**